File: EscapeUnicode.java
import java.nio.file.*;
import java.nio.charset.StandardCharsets;

public class EscapeUnicode {
    public static void main(String[] args) throws Exception {
        Files.walk(Paths.get("src/view"))
            .filter(Files::isRegularFile)
            .filter(p -> p.toString().endsWith(".java"))
            .forEach(p -> {
                try {
                    String content = new String(Files.readAllBytes(p), StandardCharsets.UTF_8);
                    StringBuilder sb = new StringBuilder();
                    boolean changed = false;
                    for (char c : content.toCharArray()) {
                        if (c > 127) {
                            sb.append(String.format("\\u%04x", (int) c));
                            changed = true;
                        } else {
                            sb.append(c);
                        }
                    }
                    if (changed) {
                        Files.write(p, sb.toString().getBytes(StandardCharsets.UTF_8));
                        System.out.println("Updated " + p);
                    }
                } catch (Exception e) {}
            });
    }
}

File: escape_unicode.py
import os
import traceback

def convert_to_escaped(filename):
    try:
        with open(filename, 'r', encoding='utf-8-sig') as f:
            content = f.read()
    except Exception as e:
        print(f"Failed to read {filename}: {e}")
        return
        
    res = []
    for c in content:
        if ord(c) > 127:
            res.append(f"\\u{ord(c):04x}")
        else:
            res.append(c)
            
    try:
        with open(filename, 'w', encoding='utf-8') as f:
            f.write(''.join(res))
        print(f"Updated {filename}")
    except Exception as e:
        print(f"Failed to write {filename}: {e}")

for root, _, files in os.walk('d:/GoiYSanPhamApp/src'):
    for file in files:
        if file.endswith('.java'):
            convert_to_escaped(os.path.join(root, file))
print("Done")
